src/react/deckgl.js

```diff
@@ -91,6 +91,17 @@ export default class DeckGL extends React.Component {
     }
     this._updateLayers(this.props);
 
+    const hasEvent = this.props.onLayerClick !== noop || this.props.onLayerHover !== noop;
+    const hasPickableLayer = this.layerManager.layers.map(l => l.props.pickable).includes(true);
+
+    if (hasEvent && !hasPickableLayer) {
+      console.warn([ //eslint-disable-line
```

Comment:
Also I can see situations where you e.g. start out with an empty DeckGL layer list and allow the user to add layers from a palette, which would trigger this warning for correct code. Keeping the message tight (a one liner in the console) might help a little.